hilti/runtime/src/types/stream.cc

```diff
@@ -173,6 +173,27 @@ int Chain::numberOfChunks() const {
     return n;
 }
 
+View View::advanceToNextData() const {
+    // Find the next offset which is not in a gap chunk.
+    auto o = 1;
+    auto c = _begin.chain()->findChunk(o + offset());
+    while ( c ) {
+        if ( ! c->isGap() )
+            return View(_begin + o, _end);
+
+        c = c->next();
+        if ( ! c )
+            break;
+
+        // Try again at the offset of the next chunk.
+        o = c->offset();
+    }
+
+    // If we haven't found a non-gap chunk simply return a view at the next
+    // byte. Since this is a gap chunk this can cause recovery in the caller.
+    return advance(1U);
```

Comment:
I find this logic hard to follow because it's mixing operation on chunks with offset-based computation. I believe it's also not the most efficient. Would it be easier to create an iterator `i = unsafeBegin(); ++i` at the beginning, then work with the chunk `c` that that's pointing to & update `i` accordingly each round; and eventually return a new view `(i, end())`?

Reply:
Simplified liked discussed out of band.